server/src/middleware/merchant-auth.ts

```diff
@@ -29,13 +38,23 @@ const merchantAuth = async (
       throw new Error('Missing Authorization header.');
 
     const [_, firebaseIdToken] = req.headers.authorization.split(' ');
-    const decodedFirebaseIdToken = await admin
-      .auth()
-      .verifyIdToken(firebaseIdToken);
-    req.decoded = decodedFirebaseIdToken;
+    const verifiedFirebaseUid = await getVerifiedUid(firebaseIdToken);
+
+    let firebaseUid;
+    if (req.body.firebaseUid !== undefined) {
+      firebaseUid = req.body.firebaseUid;
+    } else if (req.body.merchantId !== undefined) {
+      const merchant = await merchantService.getMerchant(req.body.merchantId);
+      firebaseUid = merchant.firebaseUid;
+    }
+
+    if (firebaseUid !== undefined && firebaseUid !== verifiedFirebaseUid)
+      throw new Error('Invalid bearer token.');
```

Comment:
Would prefer braces! See https://google.github.io/styleguide/jsguide.html#formatting-braces-all

```suggestion
    if (firebaseUid !== undefined && firebaseUid !== verifiedFirebaseUid) {
      throw new Error('Invalid bearer token.');
    }
```

Reply:
@hzxcaryn Oh, okay! I was thinking of single-line arrow function where no braces is used. I thought the same style applies here. But I realised that case is different because without braces it allows implicit return which is different from with braces (i.e. not just style preference but has different semantics).